webapp/channels/src/packages/mattermost-redux/src/reducers/entities/channel_categories.ts

```diff
@@ -92,7 +93,38 @@ export function byId(state: IDMappedObjects<ChannelCategory> = {}, action: Gener
 
         return changed ? nextState : state;
     }
+    case ChannelTypes.GM_CONVERTED_TO_CHANNEL: {
+        // For GM to Private channel conversion feature
+        // In the case when someone converts your GM to a private channel and moves it to a team
+        // you're not currently on, we need to remove the channel from "direct messages" category
+        // and add it to "channels" category of target team. Even though the server sends a websocket event about updated category data,
+        // it does so only for the team the channel got moved into, and not every team a user is part of, for performance reasons.
+        // For every other team, we update the state here. Everything is correct on server side, but we update state here
+        // to avoid re-fetching all categories again for all teams.
+
+        const receivedChannel = action.data as Channel;
+        const newState: IDMappedObjects<ChannelCategory> = {};
+        const categoryIDs = Object.keys(state);
```

Comment:
I wonder if it would be cleaner iterating directly on the values instead of the keys, but 0/5.

Reply:
I don't think it matters, leaving as is.